### ISSUE TITLE
Rule Change: Support test files with `max-lines-per-function`

### ISSUE THREAD

**Issue by moshest** [eslint/eslint#16000]
### What rule do you want to change?

max-lines-per-function

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
describe('my test', () => {
  describe('section 1', () => {
    it('should test1', () => {
      // ...
    });
  
    it('should test2', () => {
      // ...
    });
  });
});
```

### What does the rule currently do for this code?

count lines for each test function `it` as well as for the section function `describe`.

### What will the rule do after it's changed?

Ignore functions like `describe` and only check the `it` functions

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

We can implement this using the following new option:

```json
{
  "ignoreTestFunctions": ["describe", "describe.*"]
}
```

This will match any `EmptyFunction` expression in the following format and ignore it:

```ts
describe(name: number | string | Function | FunctionLike, fn: EmptyFunction): void;
describe.each(name: number | string | Function | FunctionLike, fn: EmptyFunction): void;
describe.foo(name: number | string | Function | FunctionLike, fn: EmptyFunction): void;
```

See [Jest type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/60b398c777a2cbfb25f81ca6f1695fc98062484e/types/jest/index.d.ts#L519) for more details.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
